Sage/carbon
# chore: update github config

### Proposed behaviour



This PR 

- removes an unused github workflow
- updates the PR template to the one I'm currently using, it simplifies the comment blocks, adds notes on codesandbox, re-orders the checklist so related items are next to each other and adds a checkbox for screenshots.

### Current behaviour


- Unused workflow due to a typo `wokflows`
- Checklist does not include screenshot

### Checklist


<del> - [ ] Screenshots are included in the PR
<del>- [ ] Carbon implementation and Design System documentation are congruent
<del>- [ ] All themes are supported

<del>- [ ] Unit tests added or updated
<del>- [ ] Cypress automation tests added or updated
<del>- [ ] Storybook added or updated
<del>- [ ] Typescript `d.ts` file added or updated

### Additional context

This was raised during our retro.

### Testing instructions

- You can view the rich diff of the PR template.
- Ensure the github actions run on this PR.
